servers/su/src/domain/clients/local_store/store.rs

```diff
@@ -855,30 +855,46 @@ impl DataStore for LocalStoreClient {
 
     /*
       Retrieve the latest message for a process.
-      Currently this is only run once for a process
-      per run of the su so it isn't very efficient
-      were pulling all the message keys into memory and
-      picking the latest one.
+      Uses reverse iteration to efficiently find the latest key
+      without loading all keys into memory.
     */
     async fn get_latest_message(
         &self,
         process_id: &str,
     ) -> Result<Option<Message>, StoreErrorType> {
-        let (paginated_keys, _) = self
-            .fetch_message_range(&process_id.to_string(), &None, &None, &None)
-            .await?;
+        let process_key_prefix = format!("message_ordering:{}:", process_id);
+        
+        // Create an upper bound for the prefix to limit iteration range
+        let mut upper_bound = process_key_prefix.clone();
+        upper_bound.push_str("\u{FFFF}");
 
-        if paginated_keys.len() < 1 {
-            return Ok(None);
-        }
+        let cf = self.index_db.cf_handle("message_ordering").ok_or_else(|| {
+            StoreErrorType::DatabaseError("Column family 'message_ordering' not found".to_string())
+        })?;
 
-        let (_, assignment_id) = match paginated_keys.last() {
-            Some(l) => l.clone(),
-            None => return Ok(None),
-        };
+        // Use reverse iterator starting from the upper bound
+        let iter = self
+            .index_db
+            .iterator_cf(
+              cf, 
+              rocksdb::IteratorMode::From(upper_bound.as_bytes(), rocksdb::Direction::Reverse)
+            );
 
-        let latest_message = self.get_message(&assignment_id)?;
+        // Find the first key that matches our prefix
+        for item in iter {
+            let (key, assignment_id_bytes) = item?;
+            let key_str = String::from_utf8(key.to_vec())?;
+            
+            if key_str.starts_with(&process_key_prefix) {
+                let assignment_id = String::from_utf8(assignment_id_bytes.to_vec())?;
+                let latest_message = self.get_message(&assignment_id)?;
+                return Ok(Some(latest_message));
+            }
+            
+            // If we've moved past our prefix, no messages exist for this process
+            break;
+        }
```

Comment:
as the loop will iterate only once, maybe an `if` would be more explicit for this logic?